homeassistant/components/epson/media_player.py

```diff
@@ -111,12 +111,13 @@ def __init__(self, projector, name, unique_id, entry):
     async def set_unique_id(self):
         """Set unique id for projector config entry."""
         _LOGGER.debug("Setting unique_id for projector.")
-        if not self._unique_id:
-            if await self._projector.get_serial_number():
-                self.hass.async_create_task(
-                    self.hass.config_entries.async_reload(self._entry.entry_id)
-                )
-                return True
+        if self._unique_id:
+            return False
+        if await self._projector.get_serial_number():
+            self.hass.async_create_task(
```

Comment:
I think we should update the config entry already here before we reload the integration. Otherwise we throw away good data and we're not sure we'll get the same data when reloading the integration. Maybe the projector turns off just after this?